omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -919,36 +751,27 @@ def _target_in_reach_of_robot_relative(self, relative_target_pose):
             target_pose (Iterable of array): Position and orientation arrays in an iterable for pose for the eef
 
         Returns:
-            bool: Whether eef can the reach target pose
+            dict: Whether each eef can reach the target pose
         """
-        return self._ik_solver_cartesian_to_joint_space(relative_target_pose) is not None
+        # TODO: output may need to be a dictionary of arms and whether they can reach the target pose
+        target_in_reach = dict()
+        for arm in self.robot.arm_names:
+            target_in_reach[arm] = self._ik_solver_cartesian_to_joint_space(relative_target_pose, arm=arm) is not None
+        return target_in_reach
 
-    @property
-    def _manipulation_control_idx(self):
+    def _manipulation_control_idx(self, arm=None):
         """The appropriate manipulation control idx for the current settings."""
-        if isinstance(self.robot, Tiago):
-            if m.TIAGO_TORSO_FIXED:
-                assert self.arm == "left", "Fixed torso mode only supports left arm!"
-                return self.robot.arm_control_idx["left"]
-            else:
-                return th.cat([self.robot.trunk_control_idx, self.robot.arm_control_idx[self.arm]])
-        elif isinstance(self.robot, Fetch):
-            return th.cat([self.robot.trunk_control_idx, self.robot.arm_control_idx[self.arm]])
+        if arm is None:
+            arm = self.arm
 
-        # Otherwise just return the default arm control idx
-        return self.robot.arm_control_idx[self.arm]
-
-    @property
-    def _manipulation_descriptor_path(self):
-        """The appropriate manipulation descriptor for the current settings."""
-        if isinstance(self.robot, Tiago) and m.TIAGO_TORSO_FIXED:
-            assert self.arm == "left", "Fixed torso mode only supports left arm!"
-            return self.robot.robot_arm_descriptor_yamls["left_fixed"]
-
-        # Otherwise just return the default arm control idx
-        return self.robot.robot_arm_descriptor_yamls[self.arm]
+        if m.TORSO_FIXED or arm == "right":
+            return self.robot.arm_control_idx[arm]
+        else:
+            # Only append trunk control idx if it is not fixed and the arm is left
+            if hasattr(self.robot, "trunk_control_idx"):
+                return th.cat([self.robot.trunk_control_idx, self.robot.arm_control_idx[arm]])
```

Comment:
Why is this needed? Also, do we actually need a separate flag `m.TORSO_FIXED`? Should we instead infer from the robot controller config?